src/tribler/core/components/bandwidth_accounting/tests/test_community.py

```diff
@@ -28,128 +30,121 @@ def create_node(self):
+        cache = self.add_cache(ID1, BandwidthTransactionSignCache(self.overlay(0), tx))
+        self.overlay(ID1).send_transaction(tx, self.address(ID2), cache.number)
 
         await self.deliver_messages()
 
-        assert self.nodes[1].overlay.database.get_total_taken(self.nodes[0].my_peer.public_key.key_to_bin()) == 0
+        assert self.database(ID2).get_total_taken(self.key_bin(ID1)) == 0
 
     async def test_ignore_unknown_transaction(self):
         """
         Test whether we are ignoring a transaction that is not in our cache.
         """
-        pk1 = self.nodes[0].my_peer.public_key.key_to_bin()
-        pk2 = self.nodes[1].my_peer.public_key.key_to_bin()
+        tx = BandwidthTransactionData(ID2, self.key_bin(ID1), self.key_bin(ID2), EMPTY_SIGNATURE, EMPTY_SIGNATURE, 1000)
+        tx.sign(self.private_key(ID1), as_a=True)
+        self.overlay(ID1).send_transaction(tx, self.address(ID2), 1234)
 
-        tx = BandwidthTransactionData(1, pk1, pk2, EMPTY_SIGNATURE, EMPTY_SIGNATURE, 1000)
-        tx.sign(self.nodes[0].my_peer.key, as_a=True)
-        self.nodes[0].overlay.send_transaction(tx, self.nodes[1].my_peer.address, 1234)
         await self.deliver_messages()
-        assert not self.nodes[0].overlay.database.get_latest_transaction(pk1, pk2)
+
+        assert not self.database(ID1).get_latest_transaction(self.key_bin(ID1), self.key_bin(ID2))
 
     async def test_concurrent_transaction_out_of_order(self):
         """
         Test creating multiple transactions, while the other party is offline and receives messages out of order.
         """
-        pk1 = self.nodes[0].my_peer.public_key.key_to_bin()
-        pk2 = self.nodes[1].my_peer.public_key.key_to_bin()
-
-        tx1 = BandwidthTransactionData(1, pk1, pk2, EMPTY_SIGNATURE, EMPTY_SIGNATURE, 1000)
-        tx2 = BandwidthTransactionData(2, pk1, pk2, EMPTY_SIGNATURE, EMPTY_SIGNATURE, 2000)
+        tx1 = BandwidthTransactionData(1, self.key_bin(ID1), self.key_bin(ID2), EMPTY_SIGNATURE, EMPTY_SIGNATURE, 1000)
+        tx2 = BandwidthTransactionData(2, self.key_bin(ID1), self.key_bin(ID2), EMPTY_SIGNATURE, EMPTY_SIGNATURE, 2000)
 
         # Send them in reverse order
-        cache = self.nodes[0].overlay.request_cache.add(BandwidthTransactionSignCache(self.nodes[0].overlay, tx1))
-        self.nodes[0].overlay.send_transaction(tx2, self.nodes[1].my_peer.address, cache.number)
+        cache = self.add_cache(ID1, BandwidthTransactionSignCache(self.overlay(ID1), tx1))
+        self.overlay(ID1).send_transaction(tx2, self.address(ID2), cache.number)
         await self.deliver_messages()
 
         # This one should be ignored by node 1
-        cache = self.nodes[0].overlay.request_cache.add(BandwidthTransactionSignCache(self.nodes[0].overlay, tx1))
-        self.nodes[0].overlay.send_transaction(tx1, self.nodes[1].my_peer.address, cache.number)
+        cache = self.add_cache(ID1, BandwidthTransactionSignCache(self.overlay(0), tx1))
```

Comment:
```suggestion
        cache = self.add_cache(ID1, BandwidthTransactionSignCache(self.overlay(ID1), tx1))
```

I forgot to convert the node id `0` into its corresponding variable.